common/auth/mapping-rule.go

```diff
@@ -26,16 +26,16 @@ import (
 )
 
 type MappingRule struct {
-	RuleName string
+	RuleName string `yaml:"RuleName"`
 
 	// Left Attribute is attribute of external user (ldap, sql, api ...)
 	// For example: displayName, mail, memberOf
-	LeftAttribute string
+	LeftAttribute string `yaml:"LeftAttribute"`
 
 	// Right Attribute is attribute of standard user
 	// For example: displayName, email
 	// Two reserved attributes: Roles, GroupPath
-	RightAttribute string
+	RightAttribute string `yaml:"RightAttribute"`
```

Comment:
Unlike `github.com/ghodss/yaml` which first converts YAML to JSON then JSON to struct [^1], `gopkg.in/yaml.v2` unmarshals using the field name lowercased as the default key [^2].

[^1]: https://pkg.go.dev/github.com/ghodss/yaml#Unmarshal
[^2]: https://pkg.go.dev/gopkg.in/yaml.v2#Unmarshal